src/components/feedback/Dialog.tsx

```diff
@@ -14,16 +15,23 @@ import type { PanelProps } from '../layout/Panel';
 type ComponentProps = {
   /**
    * Dialog _should_ be provided with a close handler. We have a few edge use
-   * cases, however, in which we need to render a "non-closeable" modal.
+   * cases, however, in which we need to render a "non-closeable" modal dialog.
    */
   onClose?: () => void;
   width?: 'sm' | 'md' | 'lg' | 'custom';
+
   /**
    * Element that should take focus when the Dialog is first rendered. When not
    * provided ("auto"), the dialog's outer element will take focus. Setting this
    * prop to "manual" will opt out of focus routing.
    */
   initialFocus?: RefObject<HTMLOrSVGElement | null> | 'auto' | 'manual';
+
+  /**
+   * Make this Dialog modal. Modal dialogs are:
+   * - Rendered with a full-screen overlay backdrop
+   */
```

Comment:
This comment list will get longer and be adjusted as more implementation is done...